zk_shell/shell.py

```diff
@@ -2712,6 +2724,61 @@ def _connect(self, hosts_list):
         else:
             self._connect_sync()
 
+    def _get_available_local_port(self):
+        """Get a random unallocated port."""
+        sock = socket.socket()
+        sock.bind(('', 0))
+        while sock.getsockname()[1] in self._tunnels_lport:
+            sock.close()
+            sock = socket.socket()
+            sock.bind(('', 0))
+        port = sock.getsockname()[1]
+        sock.close()
+        self._tunnels_lport.add(port)
+        return port
+
+    def _create_ssh_tunnel(self, rhost):
```

Comment:
oh yeah, definitely a better idea. I'll update the patch with this and address the other comments.